content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -538,4 +540,177 @@ The command will output the URL where you can access the deployed Worker.
 
 ### D1
 
-[Coming soon](https://github.com/prisma/prisma/issues/13310).
+If you are using a D1 database, you need to:
+
+- use the `@prisma/adapter-d1` database adapter (via the `driverAdapters` Preview feature)
+- set `sqlite` as the `datasource` provider in your Prisma schema
+- manually execute schema changes using [`wrangler d1 execute`](https://developers.cloudflare.com/workers/wrangler/commands/#execute) instead of using Prisma Migrate
+
+#### 1. Configure Prisma schema
+
+In your Prisma schema, add the `driverAdapters` Preview feature to the `generator` block and set the `provider` of the `datasource` to `sqlite`. If you just bootstrapped the Prisma schema with `prisma init`, also be sure to add the following `User` model to it:
+
+```prisma file=schema.prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["driverAdapters"]
+}
+
+datasource db {
+  provider = "sqlite"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id    Int     @id @default(autoincrement())
+  email String  @unique
+  name  String?
+}
+```
+
+Note that in this tutorial, you won’t need the `.env` file since the connection between Prisma ORM and D1 will happen through a [binding](https://developers.cloudflare.com/workers/configuration/bindings/).
+
+#### 2. Install dependencies
+
+Next, install the required packages:
+
+```terminal
+npm install @prisma/adapter-d1
+```
+
+#### 3. Set the D1 database connection via a binding
+
+To connect your Workers with the D1 instance, add the following binding to your `wrangler.toml` (if you don't have a D1 instance yet, you can create one using the [Cloudflare Dashboard](https://dash.cloudflare.com/) or with the [`wrangler d1 create`](https://developers.cloudflare.com/workers/wrangler/commands/#create) command):
```

Comment:
`d1 create` does not create a `wrangler.toml` afaik, just outputs the `d1_databases` part of it as the success output which can then be copied to an existing `wrangler.toml` file. Right?

Reply:
Yup, I think so too! Did anything that's written above give you the feeling that `d1 create` also creates a `wrangler.toml` that should be clarified?

Reply:
Yes, but probably just because I skimmed. The words indeed say that the database is created via that command.

Reply:
Ok, I'll leave this as is then.

Reply:
To clarify expectations:
- Cloudflare Workers users will likely already have a `wrangler.toml` file, as the default template `pnpm create cloudflare@latest <PROJECT_NAME>` provides it
- Cloudflare Pages users will likely NOT have any `wrangler.toml` file. However, to use D1 locally, they should still create their own `wrangler.toml` file, which should ONLY contain the Cloudflare bindings ([example](https://github.com/prisma/driver-adapter-qa-cf-pages/blob/063b6f626b4a7b1e559e99a3950bac698779a2fd/wrangler.toml)).